File: 1034-subarrays-with-k-different-integers/subarrays-with-k-different-integers.cpp
class Solution {
public:
     int atleast(vector<int>&nums,int k)
     {
        map<int,int>mpp;
        int j=0,count=0,n=nums.size();
        for(int i=0;i<n;i++)
        {
            mpp[nums[i]]++;
            while(mpp.size()>=k)
            {
                count+=n-i;
                mpp[nums[j]]--;
                if(mpp[nums[j]]==0)
                mpp.erase(nums[j]);
                j++;
            }
        }
        return count;
     }
    int subarraysWithKDistinct(vector<int>& nums, int k) 
    { 
        return atleast(nums,k)-atleast(nums,k+1);
    }
};

File: 1034-subarrays-with-k-different-integers/subarrays-with-k-different-integers.py
class Solution(object):
    def atleast(self, nums, k):
        count_map = {}
        j = 0
        count = 0
        n = len(nums)

        for i in range(n):
            count_map[nums[i]] = count_map.get(nums[i], 0) + 1

            while len(count_map) >= k:
                count += n - i
                count_map[nums[j]] -= 1
                if count_map[nums[j]] == 0:
                    del count_map[nums[j]]
                j += 1

        return count

    def subarraysWithKDistinct(self, nums, k):
        return self.atleast(nums, k) - self.atleast(nums, k + 1)
